### ISSUE TITLE
start_website_in_docker script fails

### ISSUE THREAD

**Issue by debanjum** [datalad/datalad#1205]
docker script failing with errors. Unable to install with `pip install -e.[full]` as sudo. 
```sh
cd [datalad_dir]/tools/testing && ./start_website_in_docker

I: installing datalad inside (in development mode)
Exception:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/pip/basecommand.py", line 122, in main
    status = self.run(options, args)
  File "/usr/lib/python2.7/dist-packages/pip/commands/install.py", line 272, in run
    InstallRequirement.from_editable(name, default_vcs=options.default_vcs))
  File "/usr/lib/python2.7/dist-packages/pip/req.py", line 115, in from_editable
    name, url, extras_override = parse_editable(editable_req, default_vcs)
  File "/usr/lib/python2.7/dist-packages/pip/req.py", line 1700, in parse_editable
    return None, url_no_extras, pkg_resources.Requirement.parse('__placeholder__' + extras).extras
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2833, in parse
    req, = parse_requirements(s)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2781, in parse_requirements
    yield Requirement(line)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2790, in __init__
    raise RequirementParseError(str(e))
RequirementParseError: Invalid requirement, parse error at "'__placeh'"
```

Installation succeeds when run in `virtualenv`.
```sh
docker exec -u 0 -it [container.name] bash
cd datalad && source venv/bin/activate
pip install -e.[full]
```

**Comment by debanjum** [comment]
Also needed to install dependencies `python-dev`, `python3-lxml` for fixing `python cryptography` and `lxml` package installations respectively

**Comment by yarikoptic** [comment]
Thanks, I will check it out.. or just send a private adding those dependencies into dockerfile

**Comment by debanjum** [comment]
Commit c350274 mentioned above maybe useful. It adds some of the dependencies but the global install of datalad in the container fails, so the commit currently installs datalad in a virtualenv.

**Comment by yarikoptic** [comment]
Should be fixed by 9efe6a924921e84d75c16a234af73336e419565b  which was pushed to master directly (no tests for this mighty script ATM :-/)

**Comment by mih** [comment]
I assume this was resolved. We can perform this type of installation in other containers also.
